src/server/lib/err.ml

```diff
@@ -0,0 +1,41 @@
+(*
+  This file is part of scilla.
+
+  Copyright (c) 2018 - present Zilliqa Research Pvt. Ltd.
+
+  scilla is free software: you can redistribute it and/or modify it under the
+  terms of the GNU General Public License as published by the Free Software
+  Foundation, either version 3 of the License, or (at your option) any later
+  version.
+
+  scilla is distributed in the hope that it will be useful, but WITHOUT ANY
+  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+  A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License along with
+  scilla.  If not, see <http://www.gnu.org/licenses/>.
+*)
+
+open Idl
+
+type t = {
+  code : int;
+  message : string;
+} [@@deriving rpcty]
+
+exception ServerError of t
+
+let rpc_err = Error.{
+  def = t;
+  raiser = (function | e -> raise (ServerError e));
+  matcher = function | ServerError e -> Some e | _ -> None
+}
+
+type reason =
+  | InvalidQuery of string
+  [@@deriving show]
+
+let mk = function
+  | InvalidQuery msg -> { code = 1; message = "Invalid query: " ^ msg }
+
+let invalid_query msg = mk (InvalidQuery msg)
```

Comment:
This code is not used (yet). The idea is to validate query parameters in the corresponding RPC handler (for example, in the `Server.runner`) and return the `InvalidQuery` error in such cases.

Reply:
Errors could be actual run-time errors such as out of gas, integer overflow etc. So the client should ultimately get the same JSON as it currently gets from scilla-runner upon failure (well, upon success too).

Reply:
This module is removed